ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/avdipaddressing.py

```diff
@@ -253,4 +247,41 @@ def vtep_ip(self) -> str:
             )
 
         offset = self._id + self._loopback_ipv4_offset
-        return self._ip(self._vtep_loopback_ipv4_pool, 32, offset, 0)
+        return get_ip_from_pool(self._vtep_loopback_ipv4_pool, 32, offset, 0)
+
+    def vrf_loopback_ip(self, pool: str) -> str:
+        """
+        Return IP address for a Loopback interface based on the given pool.
+        Default offset from pool is `id + loopback_ipv4_offset`
+
+        Used for "vtep_diagnostic.loopback".
+        """
+        offset = self.shared_utils.id + self.shared_utils.loopback_ipv4_offset
+        return get_ip_from_pool(pool, 32, offset, 0)
+
+    def evpn_underlay_l3_multicast_group(
+        self,
+        underlay_l3_multicast_group_ipv4_pool: str,
+        vrf_vni: int,
+        vrf_id: int,
+        evpn_underlay_l3_multicast_group_ipv4_pool_offset: int,
+    ) -> str:
+        """
+        Return IP address to be used for EVPN underlay L3 multicast group
+
+        TODO: Change algorithm to use VRF ID instead of VRF VNI as offset.
```

Comment:
so once this is implemented, is the `vrf_vni` param still needed? if not, should we then avoid passing both and let the caller decide which one to give?

Reply:
In case someone use a custom ip addressing class this might be needed for their logic.